geopandas/io/tests/test_parquet.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import absolute_import
+import os
+import pytest
+from pandas import DataFrame
+import numpy as np
+
+from geopandas import GeoDataFrame, GeoSeries, read_file, read_parquet
+from geopandas.array import to_wkb
+from geopandas.datasets import get_path
+from geopandas.io.parquet import (
+    _encode_metadata,
+    _decode_metadata,
+    _validate_dataframe,
+)
+from geopandas.testing import assert_geodataframe_equal, assert_geoseries_equal
+
+
+def test_encode_metadata():
+    metadata = {"a": "b"}
+
+    expected = b'{"a": "b"}'
+    assert _encode_metadata(metadata) == expected
+
+
+def test_decode_metadata():
+    metadata_str = b'{"a": "b"}'
+
+    expected = {"a": "b"}
+    assert _decode_metadata(metadata_str) == expected
+
+
+def test_validate_dataframe():
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+
+    # valid: should not raise ValueError
+    _validate_dataframe(df)
+    _validate_dataframe(df.set_index("iso_a3"))
+
+    # add column with non-string type
+    df[0] = 1
+
+    # invalid: should raise ValueError
+    with pytest.raises(ValueError):
+        _validate_dataframe(df)
+
+    with pytest.raises(ValueError):
+        _validate_dataframe(df.set_index(0))
+
+    # not a DataFrame: should raise ValueError
+    with pytest.raises(ValueError):
+        _validate_dataframe("not a dataframe")
+
+
+@pytest.mark.parametrize(
+    "test_dataset", ["naturalearth_lowres", "naturalearth_cities", "nybb"]
+)
+def test_parquet_io(test_dataset, tmpdir):
+    """Writing to parquet should not raise errors, and should not alter original
+    GeoDataFrame
+    """
+
+    df = read_file(get_path(test_dataset))
+    orig = df.copy()
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+
+    assert os.path.exists(filename)
+
+    # make sure that the original data frame is unaltered
+    assert_geodataframe_equal(df, orig)
+
+    # make sure that we can roundtrip the data frame
+    pq_df = read_parquet(filename)
+
+    assert isinstance(pq_df, GeoDataFrame)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
```

Comment:
What is the reason the `check_like` is needed here? (column order?)

Reply:
Wasn't actually needed.